test/identical.js

```diff
@@ -25,5 +26,37 @@ describe('identical', function() {
     eq(R.identical(new Number(0), 0), false);
     eq(R.identical(new Number(0), new Number(0)), false);
   });
+  
+  it('is auto-curried', function() {
+    assert.strictEqual(R.identical.length, 2);
+    var unaryFn = R.identical("foo");
+    assert.strictEqual(unaryFn.length, 1);
+    eq(unaryFn("bar"), false);
+    eq(unaryFn("foo"), true);
+    
+    eq(R.identical()("foo")()("foo"), true);
+  });
+  
+  it("does not access the placeholder property of it's arguments which is forbidden for cross-origin browser windows", function() {
+    var forbiddenPropertyAccessObject = {};
+    Object.defineProperty(
+      forbiddenPropertyAccessObject, 
+      '@@functional/placeholder', 
+      { get: function(){ throw new Error("Not allowed!"); } }
+    );
+    
+    assert.doesNotThrow(
+      () => R.identical(forbiddenPropertyAccessObject, {}),
+      Error
+    );
+    
+    assert.doesNotThrow(
+      () => R.identical({}, forbiddenPropertyAccessObject),
+      Error
+    );
+    
+    eq(R.identical(forbiddenPropertyAccessObject, forbiddenPropertyAccessObject), true);
+    eq(R.identical(forbiddenPropertyAccessObject, {}), false);
+  });
```

Comment:
I see what you're trying to do here but I've mixed feelings about the methodology. I think it's focusing a little bit too much on the implementation. If we ever decide to rename `@@functional/placeholder` this test would probably still pass and potentially create a false positive.

Perhaps an alternative would be to make sure that giving a placeholder doesn't do what it is supposed to do:

In this case we expect the placeholder to cause a function to be returned:

```javascript
add(__, 2) //=> expect a function here
```

In your case we could make sure that a placeholder is treated like any other value. Couple of options:


```javascript
assert(typeof identical(__, 42) === 'boolean');
assert(identical(__, __) === true);
assert.doesThrow(() => identical(42)(__)(42));
```

Reply:
Yes, I follow your points.

I'll integrate your idea of checking placeholder to be treated as regular value. Although that will verify `__` is no longer supported by `identical`, it will not verify the DOMException is no longer triggered. For that one would

- have to execute the test in a browser context, and setting up a cross origin iframe, or
- mock up a test object which throws on *any* property access, not just on `['@@functional/placeholder']`

To be honest, in the context of the brittle Ramda test infrastructure***, the first approach seems to me too complicated, while for the second I still haven't come up with an implementation yet.  

*** Why are the unit test written as CJS modules when the source code is all ES modules? 
Mocha can [work with ESM test files](https://mochajs.org/#nodejs-native-esm-support) for some time. Code is otherwise ES5 anyway or is it compiled to ES5 at all? 
All that `cross-env BABEL_ENV=cjs mocha --require @babel/register` seems to me much overtooled.

Reply:
Re-wrote that test with a mock object with just a few allowed properties. 

Unfortunately I had to use `Proxy` — hoping the tests don't run on Internet Explorer R.I.P. Just with ES5, I don't think a test failing for the old implementation can be written.